source/grid/grid_generator.cc

```diff
@@ -3833,12 +3837,128 @@ namespace GridGenerator
           this_cell.vertices[v] =
             cell->vertex_index(v) + triangulation_1.n_vertices();
         this_cell.material_id = cell->material_id();
+        this_cell.manifold_id = cell->manifold_id();
         cells.push_back(this_cell);
       }
 
+    // Now for the SubCellData
+    SubCellData subcell_data;
+    if (copy_manifold_ids)
+      {
+        if (dim == 2)
+          {
+            subcell_data.boundary_lines.reserve(triangulation_1.n_lines() +
+                                                triangulation_2.n_lines());
+            for (typename Triangulation<dim, spacedim>::line_iterator line =
+                   triangulation_1.begin_face();
+                 line != triangulation_1.end_face();
+                 ++line)
+              if (line->manifold_id() != numbers::flat_manifold_id)
+                {
+                  CellData<1> boundary_line;
+                  boundary_line.vertices[0] = line->vertex_index(0);
+                  boundary_line.vertices[1] = line->vertex_index(1);
+                  boundary_line.manifold_id = line->manifold_id();
+                  subcell_data.boundary_lines.push_back(
+                    std::move(boundary_line));
+                }
+            for (typename Triangulation<dim, spacedim>::line_iterator line =
```

Comment:
while not too much saving, one could add a single lambda function which takes `tria` and `offset` and call it twice on `tria_1` (offset=0) and `tria_2` (offset=n_vertces()).